scanner/src/main/kotlin/scanners/ScanCode.kt

```diff
@@ -304,7 +305,11 @@ class ScanCode(config: ScannerConfiguration) : LocalScanner(config) {
 
         val rootLicenseFile = result["files"].singleOrNull { file ->
             val path = file["path"].asText()
-            matchersForLicenseFiles.any { it.matches(Paths.get(path)) }
+            try {
+                matchersForLicenseFiles.any { it.matches(Paths.get(path)) }
+            } catch (e: InvalidPathException) {
```

Comment:
Just curious, how can the path be invalid as the path has been scanned?

Reply:
It happened once that the path in the ScanCode result contained this character:
https://www.charbase.com/dce7-unicode-invalid-character

Reply:
I still believe `InvalidPathException` shouldn't occur in the first place, but let's look into that later then.

Reply:
For the record, I couldn't reproduce the `InvalidPathException` on Windows. Seems to be something Linux / encoding specific.

Reply:
I want to check where the invalid characters come from before merging this.

Reply:
The problem occurs for example when scanning this file:
https://github.com/fshost/node-dir/tree/b40a44a74ecd8b582643088a08dab8d10c4919f6/test/fixtures/testdir5

The ScanCode result converts this filename to `testu\udce7\udcf5es.txt`. As a result we cannot open this file from Java. It could probably work if we would parse the ScanCode result using a different enconding, but then we still would not know which encoding to choose.

This problem only happens when looking for root license files, so it is safe to ignore the expection, because it is very unlikely that a root license file contains illegal characters.